lxd/images.go

```diff
@@ -2480,26 +2482,8 @@ func imageSyncBetweenNodes(d *Daemon, project string, fingerprint string) error
 		return nil
 	}
 
-	// We spread the image for the nodes inside of cluster and we need to double
-	// check if the image already exists via DB since when one certain node is
-	// going to create an image it will invoke the same routine.
-	// Also as the daily image synchronization task can be only launched by leader node,
-	// hence the leader node will have the image synced first with higher priority.
-	// In case the operation fails, the daily image synchronization task will check
-	// whether an image was synced successfully across the cluster and perform the
-	// same job if not.
-	leader, err := d.gateway.LeaderAddress()
-	if err != nil {
-		return errors.Wrap(err, "Failed to fetch the leader node address")
-	}
-
-	var targetNodeAddress string
-	if shared.StringInSlice(leader, addresses) {
-		targetNodeAddress = leader
-	} else {
-		targetNodeAddress = addresses[0]
-	}
-
+	// Pick a random node from that slice as the target.
+	targetNodeAddress := addresses[rand.Intn(len(addresses))]
```

Comment:
Please check `len(addresses)`. `rand.Intn` will panic if it's 0.

Reply:
We already do above, if addresses is <= 0 we return.

Reply:
Ah right. Didn't see that in the diff.